lib/bitstyles_phoenix/bitstyles.ex

```diff
@@ -117,36 +157,48 @@ defmodule BitstylesPhoenix.Bitstyles do
         _ -> class
       end
 
-    classname(mapping, "2.0.0")
+    class = test_only_downgrade(class, "u-version-2", "u-version-1-4")
+
+    downgrade_classname(class, target_version, {1, 5, 0})
+  end
+
+  defp downgrade_classname(class, target_version, {1, 5, 0}) do
+    # no changes when downgrading from 1.5.0 -> 1.4.0
+    downgrade_classname(class, target_version, {1, 4, 0})
   end
 
-  def classname(class, version) when version >= "1.3.0" do
-    mapping =
+  defp downgrade_classname(class, target_version, {1, 4, 0}) do
+    # downgrading 1.4.0 -> 1.3.0
+    class =
       case class do
         "u-grid-cols-" <> number -> "u-grid--#{number}-col"
         "u-col-span-" <> number -> "u-grid__col-span-#{number}"
         "u-col-start-" <> number -> "u-grid__col-#{number}"
         _ -> class
       end
 
-    classname(mapping, "1.5.0")
+    class = test_only_downgrade(class, "u-version-1-4", "u-version-1-3")
+
+    downgrade_classname(class, target_version, {1, 3, 0})
+  end
+
+  defp downgrade_classname(class, target_version, {1, 3, 0}) do
+    # no changes when downgrading from 1.3.0 -> 1.2.0
+    downgrade_classname(class, target_version, {1, 2, 0})
   end
 
-  def classname(_class, version) do
+  defp downgrade_classname(_class, target_version, {1, 2, 0}) do
     raise("""
-    The version #{version} of bitstyles is not supported. The helpers will produce incorrect classes.
+    The version #{Version.to_string(target_version)} of bitstyles is not supported. The helpers will produce incorrect classes.
     Please upgrade bitsyles and set the `bitsyles_version` to the updated version.
     """)
   end
 
-  def version do
-    bitstyles_version_override = Process.get(:bitstyles_phoenix_bistyles_version)
-
-    bitstyles_version_override ||
-      Application.get_env(:bitstyles_phoenix, :bitstyles_version, @default_version)
-  end
-
-  def default_version do
-    @default_version
+  defp test_only_downgrade(class, from, to) do
+    if Application.get_env(:bitstyles_phoenix, :add_version_test_classes, false) do
+      if(class == from, do: to, else: class)
+    else
+      class
+    end
   end
```

Comment:
```suggestion
  if Application.compile_env(:bitstyles_phoenix, :add_version_test_classes, false) do
    defp test_only_downgrade(from, from, to), do: to
    defp test_only_downgrade(class, _, _), do: class
  else
     defp test_only_downgrade(class, _, _), do: class
  end
```
This is what I meant with the compile time switch :)